gcs/upload.py

```diff
@@ -280,6 +282,135 @@ def init_write_object_grpc(cls, db, request_iterator, context):
             upload.metadata.metadata["x_emulator_no_md5"] = "true"
         return upload, is_resumable
 
+    @classmethod
+    def process_bidi_write_object_grpc(cls, db, request_iterator, context):
+        """Process a BidiWriteObject streaming RPC, and yield a stream of responses."""
+        upload, object_checksums, is_resumable = None, None, False
+        # Currently, the testbench will checkpoint and flush the data to upload.media per request for testing purposes,
+        # instead of the 15 seconds interval used in the GCS server.
+        should_checkpoint_session = True
```

Comment:
Just an idea (no action needed): maybe we should let the client pick the behavior (via headers like we do for retry testing). That way can write tests for different scenarios (e.g. when the data is not flushed and we disconnect, or when the data is flushed before the client expects it to).